### PR TITLE
Add AI_FLAG_WEIGH_ABILITY_PREDICTION

### LINKED ISSUE #5623: Use AI Rating When Guessing Abilities
### Description

`AI_DecideKnownAbilityForTurn` randomly guesses between all used ability slots on a mon. khbsd and ghoul on Discord suggested that it would be cool if we used aiRating instead, and I ALSO think it would be cool.

Maybe something like "guess ability with percentage proportional to AI rating", maybe something else, but I think this shouldn't be difficult to implement and sounds really cool. Not sure which flag if any we'd want to tie this to.

### Version

1.9.3 (Latest release)

### Upcoming/master Version

_No response_

### Discord contact info

@Pawkkie

### PR DESCRIPTION
**Pull request by Pawkkie**
## Description
There was a discussion on Discord recently that it'd be cool to have a flag that lets the AI predict the player's ability based on its aiRating, rather than just evenly distributing the predicted ability between all possible abilities. This means that the AI is more likely to assume the player will have abilities like Water Absorb rather than Damp, for example, proportionally to their aiRating.

## Issue(s) that this PR fixes
Closes #5623 

## **People who collaborated with me in this PR**
@khbsd and @ghoulslash for the idea, @AlexOn1ine and @mrgriffin for getting it working :)

## **Discord contact info**
@Pawkkie

### CODE REVIEW

**Comment by AlexOn1ine** [comment]
I find the name Predict is a bit misleading because it is the default behavior already just with weights

**Comment by Pawkkie** [comment]
> I find the name Predict is a bit misleading because it is the default behavior already just without weights

I am totally open to alternate names this was just the best I could think of

WEIGHTED_ABILITY_ASSUMPTIONS was my first draft and it's obviously a bit chonky lol

**Comment by khbsd** [comment]
maybe "GUESS"? or more specifically like "WTD_GUESS"

**Comment by AlexOn1ine** [comment]
> maybe "GUESS"? or more specifically like "WTD_GUESS"

this is also almost the same 

`AI_FLAG_WEIGH_ABILITY_PREDICTION'

**Comment by Pawkkie** [comment]
> > maybe "GUESS"? or more specifically like "WTD_GUESS"
> 
> this is also almost the same
> 
> `AI_FLAG_WEIGH_ABILITY_PREDICTION'?

would you prefer this version than? I'm down to change it :)

**Comment by AlexOn1ine** [comment]
> > > maybe "GUESS"? or more specifically like "WTD_GUESS"
> > 
> > 
> > this is also almost the same
> > `AI_FLAG_WEIGH_ABILITY_PREDICTION'?
> 
> would you prefer this version than? I'm down to change it :)

I don't mind what you end up going for but the current state is not great imo

**Comment by Pawkkie** [comment]
> > > > maybe "GUESS"? or more specifically like "WTD_GUESS"
> > > 
> > > 
> > > this is also almost the same
> > > `AI_FLAG_WEIGH_ABILITY_PREDICTION'?
> > 
> > 
> > would you prefer this version than? I'm down to change it :)
> 
> I don't mind what you end up going for but the current state is not great imo

I like `AI_FLAG_WEIGH_ABILITY_PREDICTION` I'll go with that :)

File: include/constants/battle_ai.h [hunk 1]
```diff
@@ -49,12 +49,13 @@
 #define AI_FLAG_CONSERVATIVE          (1 << 18)  // AI assumes all moves will low roll damage.
 #define AI_FLAG_SEQUENCE_SWITCHING    (1 << 19)  // AI switches in mons in exactly party order, and never switches mid-battle.
 #define AI_FLAG_DOUBLE_ACE_POKEMON    (1 << 20)  // AI has *two* Ace Pokémon. The last two Pokémons in the party won't be used unless they're the last ones remaining. Goes well in battles where the trainer ID equals to twins, couples, etc.
+#define AI_FLAG_WEIGH_ABILITY_PREDICTION       (1 << 21)  // AI will predict player's ability based on aiRating
```
Comment (Bassoonian) [inline include/constants/battle_ai.h:52]: I would realign the values so they all match

Comment (Pawkkie) [reply include/constants/battle_ai.h]: Done :)

**Pull request merged**